kobo/apps/hook/tests/hook_test_case.py

```diff
@@ -104,4 +120,47 @@ def _send_and_fail(self):
         first_hooklog = HookLog.objects.get(uid=first_hooklog_response.get("uid"))
         first_hooklog.change_status(HOOK_LOG_FAILED)
 
-        return first_hooklog_response
\ No newline at end of file
+        return first_hooklog_response
+
+    def __prepare_json_submission(self):
+        v_uid = self.asset.latest_deployed_version.uid
+        submission = {
+            "__version__": v_uid,
+            "q1": u"¿Qué tal?",
+            "group1/q2": u"¿Como está en el grupo uno la primera vez?",
+            "group1/q3": u"¿Como está en el grupo uno la segunda vez?",
+            "group2/subgroup1/q4": u"¿Como está en el subgrupo uno la primera vez?",
+            "group2/subgroup1/q5": u"¿Como está en el subgrupo uno la segunda vez?",
+            "group2/subgroup1/q6": u"¿Como está en el subgrupo uno la tercera vez?",
+            "id": self._submission_pk
+        }
+        self.__inject_submission(self.asset, submission)
+
+    def __prepare_xml_submission(self):
+        v_uid = self.asset_xml.latest_deployed_version.uid
+        submission = ("<{asset_uid}>"
+                      "   <__version__>{v_uid}</__version__>"
+                      "   <q1>¿Qué tal?</q1>"
+                      "   <group1>"
+                      "      <q2>¿Como está en el grupo uno la primera vez?</q2>"
+                      "      <q3>¿Como está en el grupo uno la segunda vez?</q3>"
+                      "   </group1>"
+                      "   <group2>"
+                      "      <subgroup1>"
+                      "          <q4>¿Como está en el subgrupo uno la primera vez?</q4>"
+                      "          <q5>¿Como está en el subgrupo uno la segunda vez?</q5>"
+                      "          <q6>¿Como está en el subgrupo uno la tercera vez?</q6>"
+                      "      </subgroup1>"
+                      "   </group2>"
+                      "   <id>{id}</id>"
+                      "</{asset_uid}>").format(
+            asset_uid=self.asset_xml.uid,
+            v_uid=v_uid,
+            id=self._submission_pk
+        )
+        self.__inject_submission(self.asset_xml, submission)
+
+    def __inject_submission(self, asset, submission):
+        self._submission_pk += 1
+        asset.deployment._mock_submission(submission)
+        asset.save(create_version=False)
```

Comment:
I don't think it's a good idea to call `asset.save()` in `_mock_submission` because it's called somewhere else in a loop. 
Instead, we can create a (public) method() that does what we want. Something like:
```
def mock_submissions(submissions):
        self.store_data({"submissions": submissions})
        self.asset.save(create_version=False)
```